src/components/create-notebook-dialog.tsx

```diff
@@ -54,6 +64,48 @@ export function CreateNotebookDialog({ children }: CreateNotebookDialogProps) {
   const [isPickerOpen, setIsPickerOpen] = useState(false);
   const router = useRouter();
   const [uploadProgress, setUploadProgress] = useState<UploadProgress>({});
+  const [selectedProvider, setSelectedProvider] = useState<string>("groq");
+  const [providerHealth, setProviderHealth] = useState<Record<string, boolean>>(
+    {
+      groq: false,
+      cerebras: false,
+    }
+  );
+
+  // Add health check effect
+  useEffect(() => {
+    const checkHealth = async () => {
+      try {
+        const providers = ["groq", "cerebras"];
```

Comment:
Since your hardcoding only these two providers - how is this adding support for provider abstraction (Read: Pick your provider feature)? 

There should be a separate list of "supported providers" which we hardcode, and then also the ability bring your own provider. In both of these we should have an abstraction class for selecting the provider.